src/rpc/blockchain.cpp

```diff
@@ -1645,34 +1646,141 @@ UniValue MempoolInfoToJSON(const CTxMemPool& pool)
     ret.pushKV("mempoolminfee", ValueFromAmount(std::max(pool.GetMinFee(maxmempool), ::minRelayTxFee).GetFeePerK()));
     ret.pushKV("minrelaytxfee", ValueFromAmount(::minRelayTxFee.GetFeePerK()));
     ret.pushKV("unbroadcastcount", uint64_t{pool.GetUnbroadcastTxs().size()});
+
+    if (feeLimits) {
+        const MempoolHistogramFeeRates& limits{feeLimits.value()};
+
+        /* Keep histogram per...
+         * ... cumulated tx sizes
+         * ... txns (count)
+         * ... cumulated fees */
+        std::vector<uint64_t> sizes(limits.size(), 0);
+        std::vector<uint64_t> count(limits.size(), 0);
+        std::vector<CAmount> fees(limits.size(), 0);
+
+        for (const CTxMemPoolEntry& e : pool.mapTx) {
+            const CAmount fee{e.GetFee()}, afees{e.GetModFeesWithAncestors()}, dfees{e.GetModFeesWithDescendants()};
+            const uint32_t size{(uint32_t)e.GetTxSize()}, asize{(uint32_t)e.GetSizeWithAncestors()}, dsize{(uint32_t)e.GetSizeWithDescendants()};
+
+            const CAmount fpb{CFeeRate{fee, size}.GetFee(1)};     // Fee rate per byte
+            const CAmount afpb{CFeeRate{afees, asize}.GetFee(1)}; // Fee rate per byte including ancestors
+            const CAmount dfpb{CFeeRate{dfees, dsize}.GetFee(1)}; // Fee rate per byte including descendants
+
+            // Fee rate per byte including ancestors & descendants
+            const CAmount tfpb{CFeeRate{afees + dfees - fee, asize + dsize - size}.GetFee(1)};
+            const CAmount fee_rate_per_byte{std::max(std::min(dfpb, tfpb), std::min(fpb, afpb))};
```

Comment:
nit: my definition of feerate is `fee per size`. This doesn't really work with `fee rate per byte` as it would be a `fee per size per size`? Also not sure about variable names like `afpb` and `dfpb`. Why not just spell it out as e.g. `ancestor_feerate` and `descendant_feerate`?


(I'm assuming `tfpb` means `total_feerate`?)

```suggestion
            const CAmount fee{e.GetFee()}, ancestor_fees{e.GetModFeesWithAncestors()}, descendant_fees{e.GetModFeesWithDescendants()};
            const uint32_t size{(uint32_t)e.GetTxSize()}, ancestor_size{(uint32_t)e.GetSizeWithAncestors()}, descendant_size{(uint32_t)e.GetSizeWithDescendants()};

            const CAmount feerate{CFeeRate{fee, size}.GetFee(1)};     // Fee rate per byte
            const CAmount ancestor_feerate{CFeeRate{ancestor_fees, ancestor_size}.GetFee(1)}; // Fee rate per byte including ancestors
            const CAmount descendant_feerate{CFeeRate{descendant_fees, descendant_size}.GetFee(1)}; // Fee rate per byte including descendants

            // Fee rate per byte including ancestors & descendants
            const CAmount total_feerate{CFeeRate{ancestor_fees + descendant_fees - fee, ancestor_size + descendant_size - size}.GetFee(1)};
            const CAmount feerate{std::max(std::min(descendant_feerate, total_feerate), std::min(feerate, ancestor_feerate))};
```

Reply:
> nit: my definition of feerate is `fee per size`. This doesn't really work with `fee rate per byte` as it would be a `fee per size per size`?

Yeah, I have the same definition of fee rate as "fee per size".

I've been trying to align with Sipa's [approach (1)](https://github.com/bitcoin/bitcoin/pull/21422#discussion_r754438676) for now.